plugins/utils/nix-develop.nix

```diff
@@ -14,9 +14,31 @@ in {
       enable = mkEnableOption "nix-develop.nvim";
 
       package = helpers.mkPackageOption "nix-develop.nvim" pkgs.vimPlugins.nix-develop-nvim;
+
+      ignoredVariables = mkOption {
+        type = types.attrsOf types.bool;
+      };
+
+      separatedVariables = mkOption {
+        type = types.attrsOf types.str;
+      };
     };
 
-  config = mkIf cfg.enable {
-    extraPlugins = [cfg.package];
-  };
+  config = let
+    inherit (cfg) ignoredVariables separatedVariables;
+  in
+    mkIf cfg.enable {
+      extraPlugins = [cfg.package];
+      extraConfigLua = ''
+        local ignored_variables = ${helpers.toLuaObject ignoredVariables}
+        for ignoredVariable, shouldIgnore in ipairs(ignored_variables) do
+        	require("nix-develop").ignored_variables[ignoredVariable] = shouldIgnore
+        end
+
+        local separated_variables = ${helpers.toLuaObject separatedVariables}
```

Comment:
please prefix the {ignored,separated}_variables with `__` to avoid risks of mixing that with an users configuration

Reply:
ahh good shout!